### DISCUSSION TITLE
Translating templates

### DISCUSSION THREAD

**Discussion by maliky** [marlowe-lang/marlowe-cardano#214] [General]
Hi,

How can I translate the three template from the https://run.marlowe-finance.io/ in french ?

I want to present this interface to french speaking African students.

**Comment by palas** [comment]
The metadata for the templates themselves is read from this folder: https://github.com/input-output-hk/marlowe-cardano/tree/main/web-common-marlowe/src/Examples/PureScript. So it could in principle be done by translating those and recompiling Marlowe Run. To translate the rest of Marlowe Run it would be necessary to look for the string literals throughout the code, and ideally we would like to add support for internationalization eventually, that would make translation easier and allow for multilanguage support in the same build.

But Marlowe Run internationalization is not in our short-term roadmap, because we are focusing on Marlowe Runtime at the moment. In fact, you may notice that source code for Marlowe Run is no longer in `main` branch (we may eventually put it in a separate repo). So, at the moment, the only way to do this would be to base any work on a previous commit like [this one](https://github.com/input-output-hk/marlowe-cardano/tree/cecc2ae962356b2d878766772725d81dcab9617e) in order to be able to build Marlowe Run.

Because of that, I wouldn't recommend spending too much time working on Marlowe Run's source at the moment, we won't be able to integrate changes to it anytime soon. Something simpler like translating just the templates for a demo, as you suggest, should be possible as I described above.
